internal/mode/static/nginx/config/servers_template.go

```diff
@@ -2,38 +2,65 @@ package config
 
 const serversTemplateText = `
 js_preload_object matches from /etc/nginx/conf.d/matches.json;
-{{ range $s := .Servers -}}
+{{ $proxyProtocol := "" }}
+{{ if $.RewriteClientIP.ProxyProtocol }}{{ $proxyProtocol = " proxy_protocol" }}{{ end }}
+
+{{- range $s := .Servers -}}
     {{ if $s.IsDefaultSSL -}}
 server {
-        {{- if or ($.IPFamily.IPv4) ($s.IsSocket) }}
-    listen {{ $s.Listen }} ssl default_server;
+         {{- if or ($.IPFamily.IPv4) ($s.IsSocket) }}
+    listen {{ $s.Listen }} ssl default_server{{ $proxyProtocol }};
         {{- end }}
         {{- if and ($.IPFamily.IPv6) (not $s.IsSocket) }}
-    listen [::]:{{ $s.Listen }} ssl default_server;
+    listen [::]:{{ $s.Listen }} ssl default_server{{ $proxyProtocol }};
         {{- end }}
-
     ssl_reject_handshake on;
+    	{{- if and ($.RewriteClientIP.ProxyProtocol) ($s.IsSocket)}}
+	set_real_ip_from unix:;
+  	    {{- else if (not $s.IsSocket)}}
+		    {{- range $cidr := $.RewriteClientIP.RealIPFrom }}
+    set_real_ip_from {{ $cidr }};
+            {{- end}}
+	    {{ end }}
+        {{- if and ($.RewriteClientIP.RealIPHeader) (not $s.IsSocket)}}
+    real_ip_header {{ $.RewriteClientIP.RealIPHeader }};
+        {{- end}}
+        {{- if and ($.RewriteClientIP.Recursive) (not $s.IsSocket)}}
+    real_ip_recursive on;
+        {{ end }}
```

Comment:
I think we can just do all of this logic in the Go code instead of having a bunch of conditionals here. Makes it easier to test and the template is cleaner.